File: python/best-time-to-buy-and-sell-stock-II.py
class Solution:
    def maxProfit(self, prices:'List[int]') -> int:
        gains = [prices[i] - prices[i-1] for i in range(1,len(prices))
                if prices[i] - prices[i-1] > 0]
        return sum(gains)
print(Solution().maxProfit([3,5,2,55,6,3,22,4,5,66,3,2,1,5,6,29]))

File: javascript/best-time-to-buy-and-sell-stock-II.js
var maxProfit = function(prices){
    let profit = 0;
    for (let i = 0; i < prices.length; i++){
        if(prices[i] > prices[i-1]){
            profit = profit + prices[i]- prices[i-1];
        }
    }
    return profit;
};

maxProfit([4,2,6,4,8,22,55,2,9])